frontend/coprs_frontend/coprs/logic/builds_logic.py

```diff
@@ -1132,15 +1132,19 @@ def finish(chroot, status):
 
                     exclusivearch = upd_dict["results"].get("exclusivearch")
                     if exclusivearch and arch not in exclusivearch:
-                        chroot.status_reason = \
-                            "This chroot was skipped because of ExclusiveArch"
-                        finish(chroot, StatusEnum("skipped"))
+                        # Skip only if all macros in ExclusiveArch evaluated
+                        if not [x for x in exclusivearch if x.startswith("%")]:
```

Comment:
And, ... coul give user a hint that something weird is happening? in the sources' builder-live.log/backend.log?

Reply:
> Macros may look like %foo a b %c, so probably we should just check for '%' in x

True, but if I understand correctly, this shouldn't matter. For example, in our beaker tests we use this specfile
https://src.fedoraproject.org/rpms/biosdevname/blob/rawhide/f/biosdevname.spec

It has `ExclusiveArch: %{ix86} x86_64` ... If it was reversed and it was `ExclusiveArch: x86_64 %{ix86}`, that would be the case you are talking about?

It shouldn't be a problem because in `copr-rpmbuild`, we produce a list out of them:

```
Output: ['exclusivearch-test-1-1.src.rpm']
Running SRPMResults tool
Package info: {
    "name": "exclusivearch-test",
    "epoch": null,
    "version": "1",
    "release": "1",
    "exclusivearch": [
        "noarch",
        "x86_64",
        "aarch64"
    ],
    "excludearch": []
}
SRPMResults finished
```

So here, in this frontend code, our `exclusivearch` variable is `["noarch", "x86_64", "aarch64"]`. If there was an unevaluated macro, it would be `["noarch", "x86_64", "aarch64", "%{zig_arches}"]`. So IMHO `startswith` should work as expected.

That being said, I have no issue with `'%' in x` and we can use it if you prefer.

> And, ... coul give user a hint that something weird is happening? in the sources' builder-live.log/backend.log?

That would be great but ... probably not? This is a frontend code, so no user-logs. And we cannot display an UI banner here either. The only indication is that at the end of the SRPM builder-live.log, there will be unevaluated macro in the `exclusivearch` list.

Reply:
You are correct. Awesome analysis. Can we have a test for this case?

Reply:
> What would happen if we produced an empty list on copr-rpmuibuild then?

That would probably be the easiest but I thought of this more as a workaround.
Anyway, I added some warnings to the builder-live.log and also tests. WDYT?